### PR TITLE
Rename class type to plugin name #api-breaking #abi-breaking

### PR DESCRIPTION
**Pull request by bmagyar**
Changes to internal API, not really an issue but I think that the message changes justify API and ABI break flags

Closes https://github.com/ros-controls/ros2_control/issues/624

### CODE REVIEW

File: controller_manager/src/controller_manager.cpp [hunk 1]
```diff
@@ -1485,7 +1485,7 @@ void ControllerManager::list_hardware_components_srv_cb(
     auto component = controller_manager_msgs::msg::HardwareComponentState();
     component.name = component_name;
     component.type = component_info.type;
-    component.class_type = component_info.class_type;
+    component.plugin_name = component_info.plugin_name;
```
Comment (destogl) [inline controller_manager/src/controller_manager.cpp:1488]: Can we deprecate this first? For not until ROS Iron?

**Review by destogl — CHANGES_REQUESTED** [review]
Fully agree, but let's deprecate it first.

File: controller_manager_msgs/msg/HardwareComponentState.msg [hunk 2]
```diff
@@ -1,6 +1,6 @@
 string name
 string type
-string class_type
+string plugin_name
```
Comment (bmagyar) [inline controller_manager_msgs/msg/HardwareComponentState.msg:3]:
```suggestion
string class_type
string plugin_name
```

File: controller_manager/src/controller_manager.cpp (hunk 1 above)
Comment (bmagyar) [reply controller_manager/src/controller_manager.cpp:1488]:
```suggestion
    component.class_type = component_info.plugin_name; // TODO: deprecated currently. Remove soon
    RCLCPP_WARN(get_logger(), "The 'class_type' field in 'controller_manager_msgs/msg/HardwareComponentState.msg' is deprecated and will be removed soon. Please switch over client code to use 'plugin_name' instead.");
    component.plugin_name = component_info.plugin_name;
```

File: hardware_interface/src/resource_manager.cpp [hunk 3]
```diff
@@ -87,18 +87,18 @@ class ResourceStorage
   {
     RCUTILS_LOG_INFO_NAMED(
       "resource_manager", "Loading hardware '%s' ", hardware_info.name.c_str());
-    // hardware_class_type has to match class name in plugin xml description
-    // TODO(karsten1987) extract package from hardware_class_type
+    // hardware_plugin_name has to match class name in plugin xml description
+    // TODO(karsten1987) extract package from hardware_plugin_name
     // e.g.: <package_vendor>/<system_type>
```
Comment (destogl) [inline hardware_interface/src/resource_manager.cpp:92]: Do we still need this comment?

```suggestion
```

**Review by destogl — DISMISSED** [review]
Just a small comments to remove additional two lines before merging

**Pull request merged**